x-pack/test/apm_api_integration/tests/alerts/helpers/alerting_api_helper.ts

```diff
@@ -88,13 +90,13 @@ export async function deleteRuleById({
   supertest,
   ruleId,
 }: {
-  supertest: SuperTestAgent;
+  supertest: SuperTestAgent | SupertestWithRoleScope;
   ruleId: string;
 }) {
   await supertest.delete(`/api/alerting/rule/${ruleId}`).set('kbn-xsrf', 'foo');
 }
 
-export async function deleteApmRules(supertest: SuperTestAgent) {
+export async function deleteApmRules(supertest: SuperTestAgent | SupertestWithRoleScope) {
   const res = await supertest.get(
     `/api/alerting/rules/_find?filter=alert.attributes.consumer:apm&per_page=10000`
   );
```

Comment:
It might be a tricky change that will complicate test failures investigations. The idea of Deployment Agnostic test design is to have its own FTR services/helpers adjusted and verified to be consistent against Kibana CI/MKI/ESS runs and be located in https://github.com/elastic/kibana/tree/main/x-pack/test/api_integration/deployment_agnostic/services

historically we were using basic authentication for API calls in Kibana tests and in DA tests we use API keys/ Cookie headers for specific SAML role. I would keep existing logic as is and use DA services for DA tests.

Reply:
Reverted those changes